scripts/generate-import-linter-config.py

```diff
@@ -0,0 +1,201 @@
+    modules: NotRequired[list[str]]
+    ignore_imports: NotRequired[list[str]]
+    unmatched_ignore_imports_alerting: NotRequired[AlertingLevel]
+    source_modules: NotRequired[list[str]]
+    forbidden_modules: NotRequired[list[str]]
+
+
+CONTRACTS = "importlinter:contracts"
+STATIC_CONFIG = {
+    "importlinter": {
+        "root_package": "ggshield",
+        "include_external_packages": True,
+    },
+    CONTRACTS: [
+        {
+            "name": "ggshield-layers",
+            "type": "layers",
+            "layers": [
+                "ggshield.cmd|*",
+                "ggshield.verticals|*",
+                "ggshield.core",
+                "ggshield.utils | pygitguardian",
+            ],
+            "ignore_imports": [
+                "ggshield.cmd.main -> ggshield.cmd.**",
+                "ggshield.cmd.** -> ggshield.cmd.utils.*",
+            ],
+            "unmatched_ignore_imports_alerting": "warn",
+        },
+        {
+            "name": "verticals-cmd-transversals",
+            "type": "forbidden",
+            "source_modules": [
+                "ggshield.cmd.*",
+            ],
+            "forbidden_modules": [
+                "ggshield.verticals.*",
+            ],
+            "ignore_imports": [
+                "ggshield.cmd.main -> ggshield.verticals.**",
+                "ggshield.cmd.{}.** -> ggshield.verticals.{}",
+                "ggshield.cmd.{}.** -> ggshield.verticals.{}.**",
+                # FIXME: #521 - enforce boundaries between cmd.auth and verticals.hmsl
+                "ggshield.cmd.auth.** -> ggshield.verticals.hmsl.**",
+            ],
+            "unmatched_ignore_imports_alerting": "none",
+        },
+    ],
+}
+
+
+def get_submodules(*, name: str, prefixed: bool) -> list[str]:
+    """Retrieve the modules inclued in a package"""
+    module = importlib.import_module(name)
+    return [
+        x.name
+        for x in pkgutil.iter_modules(
+            path=module.__path__,
+            prefix=f"{module.__name__}." if prefixed else "",
+        )
+    ]
+
+
+def expand_glob(line: str) -> list[str]:
+    """
+    Expand the *glob*
+
+    supported globs:
+    - "xxx.*" => ["xxx.sub_module_a", "xxx.sub_module_b"]
+    - "xxx|*" => ["xxx.sub_module_a | xxx.sub_module_b"]
+    - "xxx" (no globs) => ["xxx"]
+    """
+    if line.endswith(".*"):
+        return sorted(get_submodules(name=line[:-2], prefixed=True))
+    if line.endswith("|*"):
+        return [" | ".join(sorted(get_submodules(name=line[:-2], prefixed=True)))]
+    return [line]
+
+
+def expand_placeholders(line: str) -> list[str]:
+    """
+    Expand the *placeholders*
+
+    Example:
+    - "xxx.{} -> yyy.{}" => [
+        "xxx.sub_module_a -> yyy.sub_module_a",
+        "xxx.sub_module_b -> yyy.sub_module_b"
+      ]
+    """
+    if "{}" not in line:
+        return [line]
+
+    return [
+        line.replace("{}", module)
+        for module in get_submodules(
+            name=line[: line.index(".{}")],
```

Comment:
I don't think that this will increase the readability as the code is already affecting the ad-hoc value to a parameter named `name`, extracting this to a new variable named `name` and then affecting the `name` parameter with the `name` variable will not ease the reading of the code but will add one line.

But if you think is better for readability, I will do it.

Reply:
I think it would help reducing the complexity of the list comprehension. multi-line list comprehensions are a code smell to me, but that's a personal opinion. I am fine with it.